### PR TITLE
add hashed email and phone, add event_id, fix eventListener

### PR DESCRIPTION
**Pull request by ad-astra-via**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -14,6 +14,32 @@ const CLICK_ID_COOKIE = `_${CLICK_ID_PARAM}`
...
+async function hashUserData(event: MCEvent) {
+  const USER_DATA: Record<string, { hashed?: boolean }> = {
+    email_address: { hashed: true },
+    phone_number: { hashed: true },
+  }
+
+  const hashedData: Record<string, string> = {}
+  const encoder = new TextEncoder()
+
+  for (const [key, options] of Object.entries(USER_DATA)) {
```
Comment (simonabadoiu) [inline src/index.ts:26]: are we going to ever have `hashed: false` in USER_DATA? There's no need to have the options for now

File: src/index.ts [hunk 2]
```diff
@@ -39,13 +65,22 @@ const endpoints = [
...
         ...data,
         ...event.payload,
+        ...hashedUserData,
+      }
+
+      if (pageview) {
+        payload.events = '[["pageview", null]]'
+        if (!payload.event_id) {
+          payload.event_id = eventId
+        }
```
Comment (simonabadoiu) [inline src/index.ts]:
```suggestion
          payload.event_id = payload.event_id || crypto.randomUUID()
```

File: src/index.ts [hunk 3]
```diff
@@ -39,13 +65,22 @@ const endpoints = [
 const onEvent =
   (pageview = false) =>
   async (event: MCEvent) => {
+    const eventId = pageview ? crypto.randomUUID() : null
```
Comment (simonabadoiu) [inline src/index.ts:68]:
```suggestion
```

File: package.json [hunk 4]
```diff
@@ -40,10 +40,10 @@
...
     "tailwindcss": "^3.2.4",
     "ts-node": "^10.8.0",
-    "typescript": "^4.7.2",
-    "vitest": "^0.13.0"
+    "typescript": "^4.7.2"
   },
   "dependencies": {
-    "mustache": "^4.2.0"
+    "mustache": "^4.2.0",
+    "vitest": "^2.0.3"
```
Comment (simonabadoiu) [inline package.json]: why did you add vitest? are we writing tests? 😁

File: src/index.ts (hunk 1 above)
Comment (ad-astra-via) [reply src/index.ts:26]: Yes we might, for example the `p_user_id` param. As agreed on the call, for now I keep it this way, so when we pick it up in the future it will be more organised.

File: src/index.ts (hunk 2 above)
Comment (ad-astra-via) [reply src/index.ts]: In case of a pageview, we need to send the same event_id to two different endpoints. So I cannot generate UUID here, since if I do that it will be sending two different ones. If we want to get rid of the if statement, we can do:
`payload.event_id =  payload.event_id || eventId`. Is this better?

File: src/index.ts (hunk 3 above)
Comment (ad-astra-via) [reply src/index.ts:68]: This is needed so that the same event ID is sent for the two different pageview endpoints.

File: package.json (hunk 4 above)
Comment (ad-astra-via) [reply package.json]: LOL removing!

**Review by simonabadoiu — APPROVED** [review]
